File: problem-solving/baekjoon/14697_방_배정하기.go
// https://www.acmicpc.net/problem/14697v
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var a, b, c, n int
	fmt.Fscanln(reader, &a, &b, &c, &n)
	fmt.Fprintln(writer, checkRooms(a, b, c, n))
}

func checkRooms(a, b, c int, n int) int {
	for i := 0; i <= n; i++ {
		for j := 0; j <= n-i; j++ {
			for k := 0; k <= n-i-j; k++ {
				if a*i+b*j+c*k == n {
					return 1
				}
			}
		}
	}
	return 0
}

File: problem-solving/baekjoon/14697_방_배정하기.py
# https://www.acmicpc.net/problem/14697
import sys

def check_rooms(a, b, c, n):
    for i in range(0, n+1):
        for j in range(0, n+1-i):
            for k in range(0, n+1-i-j):
                if a*i + b*j + c*k == n:
                    return 1
    return 0

if __name__ == "__main__":
    a, b, c, n = list(map(int, sys.stdin.readline().split()))
    print(check_rooms(a, b, c, n))